python/gtsam/specializations/discrete.h

```diff
@@ -0,0 +1,17 @@
+/* Please refer to:
+ * https://pybind11.readthedocs.io/en/stable/advanced/cast/stl.html
+ * These are required to save one copy operation on Python calls.
+ *
+ * NOTES
+ * =================
+ *
+ * `py::bind_vector` and similar machinery gives the std container a Python-like
+ * interface, but without the `<pybind11/stl.h>` copying mechanism. Combined
+ * with `PYBIND11_MAKE_OPAQUE` this allows the types to be modified with Python,
+ * and saves one copy operation.
+ */
+
+// Seems this is not a good idea with inherited stl
```

Comment:
Yeah we have the same issue with `IndexPair` which inherits from `std::pair`. I haven't tried looking too deeply into it.

Reply:
Would be amazing if this works.

Reply:
I found this as an undocumented feature: https://github.com/pybind/pybind11/issues/620